configure.ac

```diff
@@ -93,6 +93,32 @@ if test "x$squid_host_os" = "solaris" -a "x$GCC" != "x" ; then
 	AC_USE_SYSTEM_EXTENSIONS
 fi
 
+AC_MSG_CHECKING([compiler name-version])
+squid_cv_cc_name=`$CC --version | head -1 | cut -f1 -d" "`
+AS_CASE([$squid_cv_cc_name],
+  [gcc|clang],[squid_cv_cc_majversion=`$CC -dumpversion | cut -f1 -d.`],
+  [squid_cv_cc_majversion="unknown"]
+)
+squid_cv_cc_simplified="$squid_cv_cc_name-$squid_cv_cc_majversion"
+AC_MSG_RESULT($squid_cv_cc_simplified)
+
+# warning flags
+AS_IF([test "$squid_cv_cc_simplified" = "gcc-4"],[
+    AC_MSG_NOTICE([skipping -Wextra on $squid_cv_cc_simplified])
+],[
+    # -Werror must be first
+    SQUID_CC_ADD_CXXFLAG_IF_SUPPORTED([-Werror],[squid_cv_cc_w_error])
```

Comment:
Regarding the squid_cv_cc_w_error variables. Is it too late to request a slight change that allows them to be auto-generated by the macro. So we do not need to pass in the name every time?

The auto-generate rules would be:
 * "squid_cv_cc_" prefix,
 * lowercase,
 * replace all non-alphanumerics with underscore.

We can do this as a follow. But it would be good to have it looking nicer from the start.

Reply:
I had tried that (and tried it again), it doesn't work, because the variable name is expanded into a broken shell syntax unless it's a verbatim variable name.
I'm seeing if I can push the text expansion to m4 time

Reply:
Done